gitpod-io/website
# Automatic PRs for the changelog

## Description

This PR is the next iteration for the all-mighty all-automated Changelog.

This PR adds:
- Automated creation of changelog PRs on the beginning of each month
- Updates to the Changelogger GitHub Action, which comments on PRs and adds insight into how the changelog script interpreted them
- A Slack summary script, which sends a weekly reminder about the current state of this month's changelog

## Related Issue(s)
Follow-up to #2763 

## How to test
You can execute the following in a workspace with the changes:
```
GITHUB_TOKEN=ghp_xxx node scripts/changelog/createPr.js
```

and... see what happens 😆 - it should create a new PR if a changelog PR is not already open in the repo.

See https://github.com/gitpod-io/website/pull/2924 for how this automatic PR would look like.

## Todo

- [x] Split the website repo into categories -> a mechanism for categories in other repos
- [x] When first generating a changelog file, generate a corresponding thumbnail so that CI doesn't [scream and shout](https://www.youtube.com/watch?v=kYtGl1dX5qI) https://github.com/gitpod-io/website/pull/2763#discussion_r1003328013 https://github.com/gitpod-io/website/pull/2926/commits/e8e2c3c8c65115768f0eb239058a8f1df81ea9a4
- [x] Add PR link to Slack messages https://github.com/gitpod-io/website/pull/2926/commits/93591aac4e54fa3cae535ddfd72d8fe56efddfac
- [x] Auto-linking preview environment on the changelog PRs https://github.com/gitpod-io/website/pull/2926/commits/ceb99a98d20a19cfb3b4ac5d2d9b3a198aa5ca9c
- [x] Add subcategories for workspace images and VS Code
- [x] Add a "why was this categorized as this category" section to the PR commenter https://github.com/gitpod-io/website/pull/2926/commits/16f1629ae882e34415f0faddc8a6145a2fab05b1
- [x] Add automation for Slack messages https://github.com/gitpod-io/website/pull/2926/commits/6345550e47f61bc966df2130cf10237c2b2e6a6b
- [ ] Enable excluding entries from the changelog
	1. When they are documented using partials https://github.com/gitpod-io/website/pull/2926/commits/45277ee476e8c5a6fff74c1cc9e5b8b3f0bca260
	2. When they ought to be omitted because of self-hosted releases 

## Release Notes
```release-note
NONE
```


<a href="https://gitpod-staging.com/#https://github.com/gitpod-io/website/pull/2926"><img src="https://gitpod-staging.com/button/open-in-gitpod.svg"/></a>



<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/2926"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
